frontend/src/metabase/meta/Parameter.js

```diff
@@ -451,21 +352,12 @@ export function parameterToMBQLFilter(
     return null;
   }
 
-  // $FlowFixMe: doesn't understand parameter.target[1] is a field reference
-  const fieldRef: LocalFieldReference | ForeignFieldReference =
-    parameter.target[1];
-
   if (parameter.type.indexOf("date/") === 0) {
+    const fieldRef: LocalFieldReference | ForeignFieldReference =
+      parameter.target[1];
     return dateParameterValueToMBQL(parameter.value, fieldRef);
-  } else {
-    const fieldId = FIELD_REF.getFieldTargetId(fieldRef);
-    const field = metadata.field(fieldId);
-    // if the field is numeric, parse the value as a number
-    if (isNumericBaseType(field)) {
-      return numberParameterValueToMBQL(parameter, fieldRef);
-    } else {
-      return stringParameterValueToMBQL(parameter, fieldRef);
-    }
+  } else if (is_operator(parameter.type)) {
+    return parameter_to_mbql(parameter);
```

Comment:
this uses the shared code to create mbql. Thus now the backend and frontend interpret parameters identically. This is not the case yet for dates which is why those mbql creation bits are still there. But their days are numbered